luna/router.cpp

```diff
@@ -62,8 +65,82 @@ void router::handle_request(request_method method,
                                                     validations));
 }
 
+std::string router::sanitize_path(std::string path_to_files) 
+{       
+    std::vector<std::string> url_comps;
+    std::string delimeter = "/";
+    std::string final_path = "";
+    std::string working_path = path_to_files;
+    size_t pos = 0;
+    int num_front_slash = 0, num_back_slash = 0;
+
+    // path can contain many / as prefix or suffix
+    // we want to preserve that in the final path
+    while (pos < path_to_files.size() and path_to_files[pos++] == '/') {
+        ++num_front_slash;
+    } 
+
+    pos = path_to_files.size() - 1;
+    while (int(pos) - 1 >= 0 and path_to_files[pos--] == '/') {
+        ++num_back_slash;
+    }
+
+    // Now, split by '/' to get URL components
+    while ((pos = working_path.find(delimeter)) != std::string::npos) {
+        std::string comp =  working_path.substr(0, pos);
+        if (not comp.empty()) {
+            url_comps.push_back(comp);
+        }
```

Comment:
If you push back comp regardless of whether it is empty, I think that would allow you to skip the additional checking for leading and trailing slashes with  `num_back_slash` and `num_front_slash`.

Reply:
I thought so too initially, but this logic was interfering with the stack implementation. For example, for /foo/bar/../../../confidential.txt, 

the resultant path should be the empty string since it's an invalid access. But, the because of the empty strings pushed on to the stack (if we remove that condition), we get output as confidential.txt which is not right. Enforcing the condition while posing on to the stack is the same as doing so before constructing the component array. Let me know if I'm not clear

Reply:
Makes total sense.